From 071993aac72ea8e6f9986bf41101974e8faa2eca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thomas=20Wei=C3=9Fschuh=20=28Schneider=20Electric=29?=
 <thomas.weissschuh@linutronix.de>
Date: Thu, 2 Jul 2026 11:42:03 +0200
Subject: [PATCH 03012/15333] hrtimer: Explicitly include some necessary
 headers in hrtimer_rearm.h
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Multiple used types and symbols are only visible through transitive
dependency chains.

Include the headers explicitly as those chains are going to go away.

Signed-off-by: Thomas Weißschuh (Schneider Electric) <thomas.weissschuh@linutronix.de>
Signed-off-by: Thomas Gleixner <tglx@kernel.org>
Link: https://patch.msgid.link/20260702-hrtimer-header-dependencies-v1-6-c50b19bda473@linutronix.de
---
 include/linux/hrtimer_rearm.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/linux/hrtimer_rearm.h b/include/linux/hrtimer_rearm.h
index a6f2e5d5e1c7..17a81826bd9a 100644
--- a/include/linux/hrtimer_rearm.h
+++ b/include/linux/hrtimer_rearm.h
@@ -2,7 +2,12 @@
 #ifndef _LINUX_HRTIMER_REARM_H
 #define _LINUX_HRTIMER_REARM_H
 
+#include <linux/types.h>
+
 #ifdef CONFIG_HRTIMER_REARM_DEFERRED
+#include <linux/irqflags.h>
+#include <linux/lockdep.h>
+#include <linux/preempt.h>
 #include <linux/thread_info.h>
 
 void __hrtimer_rearm_deferred(void);
-- 
2.55.0

